spec/lib/declare_schema/migration_generator_spec.rb

```diff
@@ -835,6 +835,53 @@ class Ad < ActiveRecord::Base
     expect(Ad.field_specs['company'].options[:validates].inspect).to eq("{:presence=>true, :uniqueness=>{:case_sensitive=>false}}")
   end
 
+  context 'models with the same parent foreign key relation' do
+    before do
+      class Category < ActiveRecord::Base
+        fields do
+          name :string, limit: 250, null: true
+        end
+      end
+      class Advertiser < ActiveRecord::Base
+        fields do
+          name :string, limit: 250, null: true
+        end
+        belongs_to :category, limit: 8
+      end
+      class Affiliate < ActiveRecord::Base
+        fields do
+          name :string, limit: 250, null: true
+        end
+        belongs_to :category, limit: 8
+      end
+    end
+    it 'will genereate unique constraint names' do
+      expect(Generators::DeclareSchema::Migration::Migrator.run).to(
+        migrate_up(<<~EOS.strip)
+          create_table :categories, id: :bigint, options: "CHARACTER SET utf8mb4 COLLATE utf8mb4_bin" do |t|
+            t.string :name, limit: 250, null: true, charset: "utf8mb4", collation: "utf8mb4_bin"
+          end
+          create_table :advertisers, id: :bigint, options: "CHARACTER SET utf8mb4 COLLATE utf8mb4_bin" do |t|
+            t.string  :name, limit: 250, null: true, charset: "utf8mb4", collation: "utf8mb4_bin"
+            t.integer :category_id, limit: 8, null: false
+          end
+          create_table :affiliates, id: :bigint, options: "CHARACTER SET utf8mb4 COLLATE utf8mb4_bin" do |t|
+            t.string  :name, limit: 250, null: true, charset: "utf8mb4", collation: "utf8mb4_bin"
+            t.integer :category_id, limit: 8, null: false
+          end
+          add_index :advertisers, [:category_id], name: :on_category_id
+          add_index :affiliates, [:category_id], name: :on_category_id
+          #{"add_foreign_key :advertisers, :categories, column: :category_id, name: :index_advertisers_on_category_id"}
+          #{"add_foreign_key :affiliates, :categories, column: :category_id, name: :index_affiliates_on_category_id"}
+          EOS
+      )
+      migrate
+
+      nuke_model_class(Advertiser)
+      nuke_model_class(Affiliate)
+    end
+  end if defined?(Mysql2)
```

Comment:
I think elsewhere we use `!defined?(Sqlite3)` for when to apply FKs, don't we? That's based on the assumption that Postgres will be next. It supports FKs.

Reply:
Hmm I'm getting some unexpected behavior where `Sqlite3` is always `nil`
<img width="1518" alt="Screen Shot 2021-03-19 at 1 02 50 PM" src="https://user-images.githubusercontent.com/43661219/111836713-fa9c4180-88b3-11eb-9a01-3c45315cd560.png">

Reply:
Ah, that's because of their weird capitalization where they matched their branding. It's `SQLite3`:

https://github.com/Invoca/declare_schema/blob/df444042a5d5b6d7720067cde66652485c82cd53/spec/lib/declare_schema/generator_spec.rb#L104

Reply:
🤦  Ooof. There's a couple of other spots that use the it with a lowercase. I'll go ahead and update those as well

Reply:
Oh, weird! So those places were presumably not running at all, given that casing. Hope they still pass!

Reply:
Interesting the [sqlite appraisal](https://github.com/Invoca/declare_schema/pull/69/checks?check_run_id=2152326380) is still running with the `if !defined?(SQLite3)` condition

Reply:
Oh, I wonder if it hasn't been required at that point? That was a problem with checking mysql and we wound up with this hack in a few places:
```
begin
  require 'mysql2'
  require 'active_record/connection_adapters/mysql2_adapter'
rescue LoadError
end
```

Reply:
There's also the approach of asking the driver directly (which implicitly requires the driver gems):
```
if ActiveRecord::Base.connection.class.name.match?(/SQLite3Adapter/)
```
This is the way we have to do it for actual running code (not tests) since Rails allows multiple drivers to be configured (and hence all their gems would be loaded).

Reply:
That makes a great deal of sense. Looks like it indeed had not been required at that point. Curious as to why we can't use the same driver approach in tests? I get a `ActiveRecord::ConnectionNotEstablished: No connection pool with 'primary' found.` when I try using in it in this test. Does the dummy test app just not have a connection established at this point?

Reply:
I don't fully understand when these tests are able to connect to the database. It's a very unusual setup compared to normal `rspec`. So far I've just worked around it.